File: finances/lib/data/Account.dart
class Account{
  late String _name;
  late double _balance;

  Account(this._name, this._balance);

  Account.fromJson(Map<String, dynamic> json) {
    _name = json['name'];
    _balance = json['balance'];
  }

  double get balance => _balance;

  set balance(double value) {
    _balance = value;
  }

  String get name => _name;

  set name(String value) {
    _name = value;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['name'] = _name;
    data['balance'] = _balance;
    return data;
  }
}

File: BackEnd/backend-finances-jpa/src/main/java/com/backendfinancesjpa/data/entity/Account.java
package com.backendfinancesjpa.data.entity;

import lombok.*;

import javax.persistence.*;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String name;
    private double balance;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id")
    private User user;

    public Account(String name) {
        this.name = name;
        balance = 0;
    }

}
